File: backend/streetcats-api/internal/dto/sightings.go
package dto

type SightingDTO struct {
	ID          string     `json:"id"`
	SightingID  uint64     `json:"sightingId"`
	Breed       *string    `json:"breed,omitempty"`
	Position    [2]float64 `json:"position"` // [latitude, longitude]
	Title       string     `json:"title"`
	Description string     `json:"description"`
	SpottedAt   *int64     `json:"spottedAt"`
}

type CreateSightingDTO struct {
	ID          uint64     `json:"id,omitempty"`
	AnimalID    uint64     `json:"animalId"`
	BreedID     uint64     `json:"breedId"`
	Position    [2]float64 `json:"position"` // [latitude, longitude]
	Title       string     `json:"title"`
	Description string     `json:"description"`
	SpottedAt   *int64     `json:"spottedAt"`
}

type BreedDTO struct {
	ID   uint64 `json:"id"`
	Name string `json:"name,omitempty"`
}

File: frontend/streetcats/src/types/sightings.ts

export type Sighting = {
  id: string
  sightingId: number
  title: string
  description: string
  breed: string
  position: [number, number]
  spottedAt: number | null
}

export type Breed = {
  id: number
  name: string
}

export type CreateSightingRequest = {
  animalId: number
  breedId: number
  title: string
  description: string
  position: [number, number]
  spottedAt: number | null
}

export type CreateSightingResponse = {
  id: number
  animalId: number
  breedId: number
  title: string
  description: string
  position: [number, number]
  spottedAt: number | null
}